src/main/java/com/google/firebase/database/DatabaseReference.java

```diff
@@ -177,7 +175,7 @@ public DatabaseReference push() {
    * @return The ApiFuture for this operation.
    */
   public ApiFuture<Void> setValueAsync(Object value) {
-    return new TaskToApiFuture<>(setValue(value));
+    return setValueInternal(value, PriorityUtilities.parsePriority(this.path, null), null);
```

Comment:
Are we planning on either renaming these back to setValue() or on adding blocking methods?

Reply:
There are no plans to introduce blocking methods to database, since the implementation is asynchronous all the way down to the core. Developers can wait on the futures if they need blocking-like semantics.